pkg/registry/flowcontrol/rest/storage_flowcontrol.go

```diff
@@ -71,3 +85,89 @@ func (p RESTStorageProvider) v1alpha1Storage(apiResourceConfigSource serverstora
+// if the "exempt" priority-level is not found.
+func shouldEnsureAllPredefined(flowcontrolClientSet flowcontrolclient.FlowcontrolV1alpha1Interface) (bool, error) {
+	if _, err := flowcontrolClientSet.PriorityLevelConfigurations().Get(flowcontrol.PriorityLevelConfigurationNameExempt, metav1.GetOptions{}); err != nil {
+		if apierrors.IsNotFound(err) {
+			return true, nil
+		}
+		return false, err
+	}
+	return false, nil
+}
+
+// Creates the pre-defined priority-levels and then the flow-schemas the creation are supposed to be done in
+// 30s, otherwise an error will be returned.
+func ensureAllPredefined(flowcontrolClientSet flowcontrolclient.FlowcontrolV1alpha1Interface, flowSchemas []*flowcontrolv1alpha1.FlowSchema, priorityLevels []*flowcontrolv1alpha1.PriorityLevelConfiguration) error {
+	err := wait.PollImmediate(time.Second, 30*time.Second, func() (done bool, err error) {
+		for _, flowSchema := range flowSchemas {
+			_, err := flowcontrolClientSet.FlowSchemas().Get(flowSchema.Name, metav1.GetOptions{})
+			if err != nil {
+				if apierrors.IsNotFound(err) {
+					_, err := flowcontrolClientSet.FlowSchemas().Create(flowSchema)
+					if err != nil && !apierrors.IsAlreadyExists(err) {
+						return false, err
+					} else if err == nil {
+						klog.V(3).Infof("created system preset FlowSchema %s", flowSchema.Name)
+					} else {
+						klog.V(3).Infof("system preset FlowSchema %s already exists, skipping creating", flowSchema.Name)
+					}
+				} else {
+					// Unable to get the priority class for reasons other than "not found".
+					klog.Warningf("unable to get FlowSchema %v: %v. Retrying...", flowSchema.Name, err)
+					return false, nil
+				}
+			}
+		}
+		for _, priorityLevelConfiguration := range priorityLevels {
+			_, err := flowcontrolClientSet.PriorityLevelConfigurations().Get(priorityLevelConfiguration.Name, metav1.GetOptions{})
+			if err != nil {
+				if apierrors.IsNotFound(err) {
+					_, err := flowcontrolClientSet.PriorityLevelConfigurations().Create(priorityLevelConfiguration)
+					if err != nil && !apierrors.IsAlreadyExists(err) {
+						return false, err
+					} else if err == nil {
+						klog.V(3).Infof("created system preset PriorityLevelConfiguration %s", priorityLevelConfiguration.Name)
+					} else {
+						klog.V(3).Infof("system preset PriorityLevelConfiguration %s already exists, skipping creating", priorityLevelConfiguration.Name)
+					}
+				} else {
+					// Unable to get the priority class for reasons other than "not found".
+					klog.Warningf("unable to get PriorityLevelConfiguration %v: %v. Retrying...", priorityLevelConfiguration.Name, err)
+					return false, nil
+				}
+			}
+		}
+		klog.V(4).Infof("all system flow-control settings are created successfully or already exist.")
+		return true, nil
+	})
+	// if we're never able to make it through initialization, kill the API server.
+	if err != nil {
+		return fmt.Errorf("unable to add default system flow-control settings: %v", err)
```

Comment:
i think the so-called "problem" here is mostly suggesting there's sth wrong w/ the backend etcd cluster instead of kubernetes itself given that requests from post-start-hooks are matching the exempt PL?